service/history/workflowResetor.go

```diff
@@ -141,19 +157,51 @@ func (w *workflowResetorImpl) ResetWorkflowExecution(ctx context.Context, resetR
 	}
 
 	// before changing mutable state
-	prevRunVersion := currMutableState.GetLastWriteVersion()
+	currPrevRunVersion := currMutableState.GetLastWriteVersion()
 	// terminate the current run if it is running
-	terminateCurr, closeTask, cleanupTask, retError := w.terminateIfCurrIsRunning(currMutableState, request.GetReason(), currExecution)
+	currTerminated, currCloseTask, currCleanupTask, retError := w.terminateIfCurrIsRunning(currMutableState, request.GetReason(), currExecution)
```

Comment:
Adding a new flag to reset request(terminateCurrent ) looks more natural to me. As it is also a natural behavior other API( like CLI) may need. Changing the interface is too heavy here.

Reply:
then should we expose this 'terminate' flag?

Reply:
one more thing, i cannot directly use the existing exposed API for workflow reset, since cross dc path already has the context & mutable state, and cannot simply just release the lock.

Reply:
I see your point. But having two methods with same name in this interface still looks bad to me(it is hard to understand when I comeback later ). Another option is to move the code that locking mutable states into historyEngine and let workflowResetor only does thing about reset. 

The new param in the request probably should named "terminateRunID".